tests/test_time_sync.py

```diff
@@ -69,15 +66,40 @@ def mock_adjtimex(mocker, request) -> None:
     mocker.patch("aiokatcp.adjtimex.get_adjtimex", return_value=return_value)
 
 
-def test_smoke(sensors: SensorSet, updater: TimeSyncUpdater) -> None:
-    """Test with real adjtimex, to make sure it interacts cleanly with the kernel."""
+@pytest.fixture
+def mock_no_adjtimex(mocker) -> None:
+    """Replace adjtimex with _no_adjtimex (i.e. pretend it is absent)."""
+    mocker.patch("aiokatcp.adjtimex.adjtimex", side_effect=aiokatcp.adjtimex._no_adjtimex)
+
+
+_linux = pytest.mark.skipif(
+    sys.platform != "linux",
+    reason="Check real adjtimex on Linux systems only",
+)
+_non_linux = pytest.mark.skipif(
+    sys.platform == "linux",
+    reason="Check unimplemented adjtimex on non-Linux systems only",
+)
+
+
+@pytest.mark.parametrize(
+    "expected_status",
+    [
+        pytest.param(Sensor.Status.NOMINAL, marks=_linux),
+        pytest.param(Sensor.Status.INACTIVE, marks=_non_linux),
+    ],
+)
+def test_smoke(
+    sensors: SensorSet, updater: TimeSyncUpdater, expected_status: Sensor.Status
+) -> None:
+    """Test with real adjtimex on Linux to ensure it interacts cleanly with kernel."""
```

Comment:
Not sure that docstring is accurate. It only covers one of the two parametrisations.

Come to think of it, I'm not sure parametrisation is really necessary, since they're a partition. You could just write
```python
expected_status = Sensor.Status.NOMINAL if sys.platform == "linux" else Sensor.Status.INACTIVE
```
inside the test function.

Reply:
Thanks, I missed the partition. I originally had tests for linux and darwin until I merged them.